share/completions/magento.fish

```diff
@@ -0,0 +1,720 @@
+  echo MyISAM\t"MyISAM storage engine"
+  echo InnoDB\t"Supports transactions, row-level locking, and foreign keys"
+  echo BLACKHOLE\t"/dev/null storage engine (anything you write to it disappears)"
+  echo MEMORY\t"Hash based, stored in memory, useful for temporary tables"
+  echo CSV\t"CSV storage engine"
+  echo ARCHIVE\t"Archive storage engine"
+end
+
+###
+
+###
+# Retrieves all available MySQL users
+###
+function __fish_print_magento_mysql_users -d "Shows all available MySQL users"
+    set -l users (__fish_print_magento_mysql_command 'select user from mysql.user;' | sort -u)
+
+    for i in $users
+      echo $i\t"$i"
+    end
+end
+
+###
+# Retrieves all available MySQL databases
+###
+function __fish_print_magento_mysql_databases -d "Shows all available MySQL databases"
+    set -l tables (__fish_print_magento_mysql_command 'show databases;')
+
+    for i in $tables
+      echo $i\t"$i"
+    end
+end
+
+function __fish_print_magento_modules -d "Lists all Magento modules"
+  set -l modules (magento module:status)
+
+  for i in $test
+    if [ (echo $i) -a (echo $i) != 'None' ]
+      echo $i 
+    end
+  end
+end
+
+function __fish_magento_print_indexer_modes -d "Lists all indexer modes"
+  echo realtime\t"Index in realtime"
+  echo schedule\t"Index via cron jobs"
+end
+
+###
+# Retrieves all existing magento users from the mysql database
+###
+function __fish_print_magento_indexer_names -d "Shows all available magento themes"
+    set -l indexers (__fish_print_magento_mysql_command 'select indexer_id from indexer_state;')
+
+    for i in $indexers
+      echo $i\t"$i"
+    end
+end
+
+function __fish_print_magento_i18n_packing_modes -d "Shows all available packing modes"
+  echo replace\t"replace language pack by new one"
+  echo merge\t"merge language packages"
+end
+
+function __fish_print_magento_available_ides -d "Shows all IDEs supported by magento"
+  echo phpstorm\t"JetBrains PhpStorm"
+end
+
+function __fish_print_magento_available_tests -d "Shows all available magento tests"
+  echo all\t"all tests"
+  echo unit\t"unit tests only"
+  echo integration\t"integration tests only"
+  echo integration-all\t"all integration tests"
+  echo static\t"static file tests only"
+  echo static-all\t"all static file tests"
+  echo integrity\t"integrity tests only"
+  echo legacy\t"legacy tests only"
+  echo default\t"all tests"  
+end
+
+###
+# Retrieves all existing magento themes from the mysql database
+###
+function __fish_print_magento_themes -d "Shows all available magento themes"
+	  set -l themes (__fish_print_magento_mysql_command 'select theme_path from theme;')
+
+    for i in $themes
+      echo $i\t"$i"
+    end
+end
+
+function __fish_print_magento_theme_areas -d "Shows all available magento theme areas"
+	echo frontend\t"Frontend"
+	echo adminhtml\t"Backend"
+end
+
+function __fish_print_magento_languages -d "Shows all existing magento languages"
+	echo af_ZA\t"Afrikaans (South Africa)"
```

Comment:
Isn't there a way to generate this programatically? Perhaps this corresponds to files somewhere?